src/unix/fs.c

```diff
@@ -739,6 +734,45 @@ static int uv__fs_fstat(int fd, uv_stat_t *buf) {
 }
 
 
+typedef ssize_t (*uv__fs_buf_iter_yield)(uv_fs_t* req);
+static ssize_t uv__fs_buf_iter(uv_fs_t* req, uv__fs_buf_iter_yield yield) {
+  unsigned int iovmax;
+  unsigned int nbufs;
+  uv_buf_t* bufs;
+  ssize_t total;
+  ssize_t result;
+
+  iovmax = uv__getiovmax();
+  nbufs = req->nbufs;
+  bufs = req->bufs;
+  total = 0;
+
+  while (nbufs) {
+    req->nbufs = nbufs;
+    if (req->nbufs > iovmax) {
+      req->nbufs = iovmax;
+    }
+
+    result = yield(req);
+    if (result <= 0) {
+      total = result;
```

Comment:
Callers should already be prepared to deal with partial reads and writes so that should not be an issue.

The problem with retrying on error is that you don't know how much data has been sent.  With reads, it's easy to miss that the contents of your receive buffers may have been clobbered and that's a potential security liability.